oxide-auth-actix/examples/actix-example/src/main.rs

```diff
@@ -0,0 +1,236 @@
+use futures::{future, Future};
+use oxide_auth::{
+    endpoint::{Endpoint, OAuthError, OwnerConsent, OwnerSolicitor, PreGrant, Scopes, Template},
+    frontends::simple::endpoint::{EndpointErrorMapper, FnSolicitor, Generic},
+    primitives::prelude::{
+        AuthMap, Authorizer, Client, ClientMap, Issuer, RandomGenerator, Registrar, Scope, TokenMap,
+    },
+};
+use oxide_auth_actix::{
+    Authorize, OAuthMessage, OAuthOperation, OAuthRequest, OAuthResource, OAuthResponse, Refresh,
+    Resource, Token, WebError,
+};
+use std::thread;
+
+static DENY_TEXT: &str = "<html>
+This page should be accessed via an oauth token from the client in the example. Click
+<a href=\"http://localhost:8020/authorize?response_type=code&client_id=LocalClient\">
+here</a> to begin the authorization process.
+</html>
+";
+
+struct State {
+    registrar: ClientMap,
+    authorizer: AuthMap<RandomGenerator>,
+    issuer: TokenMap<RandomGenerator>,
+    scopes: Vec<Scope>,
+}
+
+type Extras = Option<Box<dyn OwnerSolicitor<OAuthRequest> + Send + 'static>>;
+
+/// Example of a main function of an actix-web server supporting oauth.
+pub fn main() {
+    std::env::set_var(
+        "RUST_LOG",
+        "actix_example=info,actix_web=info,actix_http=info,actix_service=info",
+    );
+    env_logger::init();
+
+    let mut sys = actix::System::new("HttpServerClient");
+
+    let state = State::preconfigured().start();
+
+    // Create the main server instance
+    HttpServer::new(move || {
+        App::new()
+            .data(state.clone())
+            .wrap(Logger::default())
+            .service(
+                web::resource("/authorize")
+                    .route(web::get().to_async(
+                        |(req, state): (OAuthRequest, web::Data<Addr<State>>)| {
+                            state
+                                .send(Authorize(req).wrap(Some(Box::new(FnSolicitor(
+                                    move |_req: &mut OAuthRequest, pre_grant: &PreGrant| {
+                                        // This will display a page to the user asking for his permission to proceed. The submitted form
+                                        // will then trigger the other authorization handler which actually completes the flow.
+                                        OwnerConsent::InProgress(
+                                            OAuthResponse::ok()
+                                                .content_type("text/html")
+                                                .unwrap()
+                                                .body(&crate::support::consent_page_html(
+                                                    "/authorize".into(),
+                                                    pre_grant,
+                                                )),
+                                        )
+                                    },
+                                )))))
```

Comment:
Instead of sending the compete box here, we could also send a simple enum variant.

```
enum OwnerInteraction {
    NonInteractiveRequest,
    Post(String),
}
```

Then the `State` can construct the correct boxed solicitor in its `Handler` implementation itself, which would fit its role better. This also removes the need to typedefing the `Extras`.

Reply:
The `GET` handler here could also have a comment how that request could be replayed. In particular, only `POST` is Cors protected by default. I think that would be instructive.

Reply:
Oh right. The enum could contain three variants so as to avoid the need of wrapping it into a `Option`.

Reply:
What sort of comment should be there? just like `// Watch out! Requests for this resource are repeatable`

Reply:
Unfortunately the `post` is of course not protected perfectly against CSRF as well but better. So something along the lines of:

> GET requests should not mutate server state and are extremely vulnerable accidental repetition as well as Cross-Site Request Forgery (CSRF).